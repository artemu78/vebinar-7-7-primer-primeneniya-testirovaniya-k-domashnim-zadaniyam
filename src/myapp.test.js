const TODO_LIST = require("./assets/stabbed_todo_list.json");
const { makeDomElement } = require("./todo");
const { readCache, saveCache } = require("./cache");
const { request } = require("./service");

const todoListStringified = JSON.stringify(TODO_LIST);
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(TODO_LIST),
  }),
);

describe("Тесты веб приложения", () => {
  test("запись в кэш", () => {
    Storage.prototype.setItem = jest.fn();
    saveCache(TODO_LIST);

    expect(localStorage.setItem).toHaveBeenCalledWith(
      "todos",
      todoListStringified,
    );
    expect(localStorage.setItem).toHaveBeenCalledTimes(1);
  });

  test("чтение пустого кэша", () => {
    Storage.prototype.getItem = jest.fn().mockReturnValue(null);
    const cache = readCache();
    expect(cache).toStrictEqual(null);
    expect(localStorage.getItem).toHaveBeenCalledWith("todos");
  });

  test("чтение кэша с данными", () => {
    Storage.prototype.getItem = jest.fn().mockReturnValue(todoListStringified);
    const cache = readCache();
    expect(cache).toStrictEqual(TODO_LIST);
    expect(localStorage.getItem).toHaveBeenCalledWith("todos");
  });

  test("создание DOM элемента", () => {
    const result = makeDomElement(TODO_LIST[0]);
    expect(result).toMatchSnapshot();
    expect(result.tagName).toBe("DIV");
  });

  test("запрос к сети", async () => {
    const serviceResult = await request();
    expect(serviceResult).toBe(TODO_LIST);
  });
});
