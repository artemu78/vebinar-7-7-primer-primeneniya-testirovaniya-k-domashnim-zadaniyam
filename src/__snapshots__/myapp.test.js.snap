// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты веб приложения создание DOM элемента 1`] = `<div />`;

exports[`Тесты нашего приложения создание DOM элемента 1`] = `<div />`;
